modules/siddhi-extensions/regex/src/test/java/org/wso2/siddhi/extension/regex/FindFunctionExtensionTestCase.java

```diff
@@ -0,0 +1,288 @@
+    }
+
+    @Test
+    public void testFindStartFromIndexFunctionExtensionTestCase() throws InterruptedException {
+        log.info("FindStartFromIndexFunctionExtension TestCase");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "define stream inputStream (symbol string, price long, regex string, "
+                + "startingIndex int);";
+        String query = ("@info(name = 'query1') " +
+                "from inputStream " +
+                "select symbol , regex:find(regex, symbol, startingIndex) as aboutWSO2 " +
+                "insert into outputStream;");
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime
+                (inStreamDefinition + query);
+
+        executionPlanRuntime.addCallback("query1", new QueryCallback() {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+                for (Event inEvent : inEvents) {
+                    count++;
+                    if (count == 1) {
+                        Assert.assertEquals(true, inEvent.getData(1));
+                    }
+                    if (count == 2) {
+                        Assert.assertEquals(false, inEvent.getData(1));
+                    }
+                    eventArrived = true;
+                }
+            }
+        });
+
+        InputHandler inputHandler = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.start();
+        inputHandler.send(new Object[]{"21 products are produced within 10 years by WSO2 currently "
+                                               + "by WSO2 employees", 60.5f, "\\d\\d(.*)WSO2", 30});
+        inputHandler.send(new Object[]{"21 products are produced within 10 years by WSO2 currently "
+                                               + "by WSO2 employees", 60.5f, "\\d\\d(.*)WSO2", 35});
+
+        Thread.sleep(100);
+        Assert.assertEquals(2, count);
+        Assert.assertTrue(eventArrived);
+        executionPlanRuntime.shutdown();
+    }
+
+    @Test (expected = ExecutionPlanValidationException.class)
+    public void testFindFunctionExtensionTestCase1() throws InterruptedException {
+        log.info("FindFunctionExtension TestCase with invalid number of arguments");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "define stream inputStream (symbol string, price long, regex string);";
+        String query = ("@info(name = 'query1') " +
+                "from inputStream " +
+                "select symbol , regex:find(regex) as aboutWSO2 " +
+                "insert into outputStream;");
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime
+                (inStreamDefinition + query);
+
+        executionPlanRuntime.start();
```

Comment:
You don't need to start the execution plan for parameter validation scenarios

Reply:
Fix for this for all other places

Reply:
fixed in 759a011